src/main/java/com/ibm/nfvodriver/driver/NSDManagementDriver.java

```diff
@@ -243,9 +253,10 @@ public byte[] getNsdArchiveContentDescriptor(final ResourceManagerDeploymentLoca
      */
     public String putNsdArchiveContentDescriptor(final ResourceManagerDeploymentLocation deploymentLocation, String nsdInfoId, byte[] nsdArchiveContent) throws SOL005ResponseException {
         final String url = deploymentLocation.getProperties().get(NFVO_SERVER_URL) + API_CONTEXT_ROOT + API_PREFIX_NS_DESCRIPTORS + "/" + nsdInfoId + API_NSD_CONTENT;
-        final HttpHeaders headers = getHttpHeaders(deploymentLocation);
+        final HttpEntity<byte[]> requestEntity = constructHttpByteHeader(deploymentLocation, MediaType.APPLICATION_JSON, nsdArchiveContent);
```

Comment:
instead of constructHttpByteHeader(xxxxx) function. use the overloaded createRequestEntity(xxx).